Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -22,11 +22,27 @@ YUI.add('ez-roleserversideviewservice', function (Y) {
     Y.eZ.RoleServerSideViewService = Y.Base.create('roleServerSideViewService', Y.eZ.ServerSideViewService, [], {
         initializer: function () {
             this.on('*:contentDiscover', function (e) {
-                e.config.contentDiscoveredHandler = Y.bind(this._assignRole, this);
+                // If we're not discovering for a role assignment, the contentDiscoveredHandler is already set
+                if (this._isDiscoveringForAssigningRole(e)) {
```

Comment:
@yannickroger This is what it does - or rather, it removes a parameter, and checks whether it is absent. If I added a boolean for this I could reduce `return data !== undefined && data.roleId !== undefined;` to `return data.isAssigningRole;` - is it worth it? (Did I understand you correctly?)